src/win.c

```diff
@@ -1887,10 +1889,14 @@ bool destroy_win_start(session_t *ps, struct win *w) {
 	}
 
 	if (w->managed) {
-		// Clear PIXMAP_STALE flag, since the window is destroyed there is no
-		// pixmap available so STALE doesn't make sense.
+		// Clear IMAGES_STALE flags since the window is destroyed: Clear
+		// PIXMAP_STALE as there is no pixmap available anymore, so STALE doesn't
+		// make sense.
+		// XXX Clear SHADOW_STALE as setting/clearing flags on a destroyed window
```

Comment:
Maybe we could loose that check?

Reply:
Or maybe we should actually regenerate the shadow image here, as we clear the stale flag?

Reply:
We are destroying the window at this point. We could "manually" recreate the shadow here if the flag is set. Updating the flags should work, as long as we haven't set `w->state = WSTATE_DESTROYING`.
Is there a specific reason the image updates are only done in the x critical section, that would be violated if we call the update here?

Reply:
I see, that make sense. Yeah, I think this is fine